python/mlx/optimizers.py

```diff
@@ -52,6 +30,25 @@ def update(self, model: "mlx.nn.Module", gradients: dict):
         """
         model.update(self.apply_gradients(gradients, model))
 
+    def init(self, model: dict):
+        """Initialize the optimizer's state
+
+        Args:
+            model (dict): A Python tree of parameters
```

Comment:
It is unclear if this should be parameters or gradients especially given line 66.

Reply:
Well really it can be either, In fact it just needs to be a tree that has the same shape / type as the gradients.

But the typical usage would be `optimizer.init(model.trainable_parameters())`

Reply:
I will make it more clear in the doc